package/src/components/MessageList/MessageList.tsx

```diff
@@ -485,19 +530,55 @@ const MessageListWithContext = <
    * Effect to mark the channel as read when the user scrolls to the bottom of the message list.
    */
   useEffect(() => {
-    const listener: ReturnType<typeof channel.on> = channel.on('message.new', (event) => {
-      const newMessageToCurrentChannel = event.cid === channel.cid;
-      const mainChannelUpdated = !event.message?.parent_id || event.message?.show_in_channel;
+    const shouldMarkRead = () => {
+      return (
+        !channelUnreadState?.first_unread_message_id &&
+        !threadList &&
```

Comment:
Just curious here, why do we skip threads in this ?

Reply:
We don't have unread counts supported for threads from BE.

Reply:
React doesn't support it, either. To be honest, we have never worked on thread unread counts. We even hide the action of marking threads as unread.

Reply:
We have them in the `ThreadList` for example; please check if everything works in order there (it should, but just to be sure)